WPFHospitalEditor/MapObject/MapObject.cs

```diff
@@ -0,0 +1,96 @@
+        }
+        public void setTextBlockPositionOnMap(MapObjectCoordinates mapObjectCoordinates) {
+            this.name.SetValue(Canvas.LeftProperty, mapObjectCoordinates.X);
+            this.name.SetValue(Canvas.TopProperty, mapObjectCoordinates.Y); 
+        }
+        public void setMapObjectColor()
+        {
+            MapObjectColors MapObjectColors = new MapObjectColors();
+            this.rectangle.Fill = MapObjectColors.MapObjectTypesColors[mapObjectType];
+            if (mapObjectType != MapObjectType.parking && mapObjectType != MapObjectType.road && mapObjectType != MapObjectType.waitingRoom && mapObjectType != MapObjectType.parkingSlot)
+            {
+                this.rectangle.Stroke = Brushes.Black;
+            }
+        }
+        public void setMapObjectDoor(MapObjectDoorOrientation mapObjectDoorOrientation)
+        {
+            MapObjectCoordinates mocDoor = new MapObjectCoordinates(0, 0);
+            if(mapObjectDoorOrientation == MapObjectDoorOrientation.up)
+            {
+                this.mapObjectDoor.rectangle.SetValue(Canvas.LeftProperty, mapObjectMetrics.mapObjectCoordinates.X + mapObjectMetrics.mapObjectDimensions.Width / 2 - AllConstants.doorWidth / 2);
+                this.mapObjectDoor.rectangle.SetValue(Canvas.TopProperty, mapObjectMetrics.mapObjectCoordinates.Y - AllConstants.doorHeight / 2);
+            } else if(mapObjectDoorOrientation == MapObjectDoorOrientation.down)
+            {
+                this.mapObjectDoor.rectangle.SetValue(Canvas.LeftProperty, mapObjectMetrics.mapObjectCoordinates.X + mapObjectMetrics.mapObjectDimensions.Width / 2 - AllConstants.doorWidth / 2);
+                this.mapObjectDoor.rectangle.SetValue(Canvas.TopProperty, mapObjectMetrics.mapObjectCoordinates.Y + mapObjectMetrics.mapObjectDimensions.Height - AllConstants.doorHeight / 2);
+            } else if(mapObjectDoorOrientation == MapObjectDoorOrientation.left)
+            {
+                this.mapObjectDoor.rectangle.SetValue(Canvas.LeftProperty, mapObjectMetrics.mapObjectCoordinates.X - AllConstants.doorHeight / 2);
+                this.mapObjectDoor.rectangle.SetValue(Canvas.TopProperty, mapObjectMetrics.mapObjectCoordinates.Y + mapObjectMetrics.mapObjectDimensions.Height / 2 - AllConstants.doorWidth / 2);
+            } else if(mapObjectDoorOrientation == MapObjectDoorOrientation.right)
+            {
+                this.mapObjectDoor.rectangle.SetValue(Canvas.LeftProperty, mapObjectMetrics.mapObjectCoordinates.X + mapObjectMetrics.mapObjectDimensions.Width - AllConstants.doorHeight / 2);
+                this.mapObjectDoor.rectangle.SetValue(Canvas.TopProperty, mapObjectMetrics.mapObjectCoordinates.Y + mapObjectMetrics.mapObjectDimensions.Height / 2 - AllConstants.doorWidth / 2);
+            } else
+            {
+                this.mapObjectDoor.rectangle.Visibility = Visibility.Hidden;
+            }
+        }
```

Comment:
Depending on door orientation (mapObjectDoorOrientation), this method sets coordinates of the door in the middle of object side(up/down/left/right). Maybe we can use 'switch-case' instead of 'if-else' to simplify this. If it still looks complicated we will try to make it more simple.